server/sonar-web/src/main/js/apps/overview/stickers/StickersModal.tsx

```diff
@@ -22,30 +22,39 @@ import Modal from '../../../components/controls/Modal';
 import StickerButton from './StickerButton';
 import StickerSnippet from './StickerSnippet';
 import StickerParams from './StickerParams';
-import { translate } from '../../../helpers/l10n';
 import { getStickerUrl, StickerType, StickerOptions } from './utils';
+import { Component } from '../../../app/types';
+import { translate } from '../../../helpers/l10n';
 import './styles.css';
 
+interface Props {
+  component: Component;
+}
+
 interface State {
   open: boolean;
   selectedType: StickerType;
   stickerOptions: StickerOptions;
 }
 
-export default class StickersModal extends React.PureComponent<{}, State> {
-  mounted: boolean;
-  state: State = {
-    open: false,
-    selectedType: StickerType.marketing,
-    stickerOptions: { color: 'white' }
-  };
-
-  componentDidMount() {
-    this.mounted = true;
+export default class StickersModal extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      open: false,
+      selectedType: StickerType.measure,
+      stickerOptions: {
+        color: 'white',
+        component: props.component.key,
```

Comment:
don't store the `branch` and `component` fields in `state` because they're easy retrievable from `props`